allow custom comment type to fetch avatar urls in comment api

rishishah-multidots opened the pull request:
### Pull Request Description
Title: Allow Avatar URL for Custom Comment Type in Comment API

### Overview:
This PR introduces a new feature that enables the inclusion of an avatar URL for custom comment types in the WordPress Comment API. This enhancement allows developers to specify custom avatar URLs, improving the flexibility and usability of the comment system for custom comment types.

### Changes Made:

- Updated the Comment API to accept an avatar_url parameter for custom comment types.

### Why This Matters:

- If any custom comment type is passed on comment API then avatar URLs is not fetching so need to add support avatar URLs for custom comment type.

### Related Tickets:
[Ticket: #60622](https://github.com/WordPress/gutenberg/pull/60622): Discussion on block comments feature.

rishishah-multidots committed:
allow custom comment type to fetch avatar urls in comment api

ellatrix commented on src/wp-includes/link-template.php:
```diff
@@ -4341,9 +4341,9 @@ function is_avatar_comment_type( $comment_type ) {
+	 * @param array $types An array of content types.
 	 */
-	$allowed_comment_types = apply_filters( 'get_avatar_comment_types', array( 'comment' ) );
+	$allowed_comment_types = apply_filters( 'get_avatar_comment_types', array( '' ) );
```
Why does this need modification at all? We're already filtering it in the GB PR?

rishishah-multidots replied:
Since we generated the comment types PR, this modification ensures consistency and helps prevent potential issues in the future.

rishishah-multidots committed:
Update avatar.php

made changes for the avatar url in the get comment rest API test cases.

rishishah-multidots committed:
made changes to allow custom comment type to fetch avatar URLs in the comment API

made changes to allow custom comment type to fetch avatar URLs in the comment API

t-hamano commented:
The Block-level comments functionality is still an experimental feature in the Gutenberg plugin, so there's no need to backport it to core. In fact, the backport changelog has been removed.

https://github.com/WordPress/gutenberg/pull/68846

I'm sorry, but let me close this PR. We need to continue to iterate on the Gutenberg plugin and create a backport PR once we have decided to "stabilize" the feature.
